File: src/ndi/+ndi/+cloud/+sync/+internal/+index/getIndexFilepath.m
function indexPath = getIndexFilepath(ndiDatasetPath, mode, options)
    arguments
        ndiDatasetPath (1,1) string {mustBeFolder}
        mode (1,1) string {mustBeMember(mode, ["read", "write"])}
        options.Verbose (1,1) logical = true
    end

    syncDirPath = fullfile(ndiDatasetPath, '.ndi', 'sync');
    if ~isfolder(syncDirPath)
        if mode == "write"
            if options.Verbose
                fprintf('Creating sync directory: %s\n', syncDirPath);
            end
            mkdir(syncDirPath);
        end
    end
    indexPath = fullfile(syncDirPath, 'index.json');
end

File: ndi-python/ndi/cloud/sync/internal/index/get_index_filepath.py
"""Get the filepath for the sync index file.

Ported from: ndi.cloud.sync.internal.index.getIndexFilepath (MATLAB)
"""

import os
from typing import Literal


def get_index_filepath(
    ndi_dataset_path: str,
    mode: Literal["read", "write"],
    verbose: bool = True
) -> str:
    """Get the filepath for the sync index file.

    Args:
        ndi_dataset_path: Path to the NDI dataset (must be a folder)
        mode: Whether to get path for reading or writing
        verbose: If True, print verbose output

    Returns:
        Path to the sync index file

    Raises:
        ValueError: If ndi_dataset_path is not a directory
    """
    if not os.path.isdir(ndi_dataset_path):
        raise ValueError(f"ndi_dataset_path must be a folder: {ndi_dataset_path}")

    sync_dir_path = os.path.join(ndi_dataset_path, '.ndi', 'sync')

    if not os.path.isdir(sync_dir_path):
        if mode == "write":
            if verbose:
                print(f'Creating sync directory: {sync_dir_path}')
            os.makedirs(sync_dir_path, exist_ok=True)

    index_path = os.path.join(sync_dir_path, 'index.json')
    return index_path
